docs/madr/decisions/029-kds-sync-hash-suffix.md

```diff
@@ -0,0 +1,117 @@
+# Use hash-suffix for policy naming during KDS sync
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/3837
+
+## Context and Problem Statement
+
+Primary keys are different for resources on Universal and Kubernetes:
+
+* Universal: (name, namespace, mesh, type)
+* Kubernetes: (name, namespace, kind)
+
+This results in inconsistent state when Global(Universal) and Zone(Kubernetes):
+
+* Global(Universal):
+```yaml
+type: MeshTrafficPermission
+name: allow-all
+mesh: mesh-1
+spec: {}
+---
+type: MeshTrafficPermission
+name: allow-all
+mesh: mesh-2
+spec: {}
+```
+
+* Zone(Kubernetes):
+```yaml
+kind: MeshTrafficPermission
+metadata:
+  name: allow-all
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: mesh-1
+spec: {}
+```
+
+MeshTrafficPermission from `mesh-2` is not synced to Zone because there is already MeshTrafficPermission with 
+a key `(allow-all, kuma-system, MeshTrafficPermission)`.
+
+## Considered Options
+
+* use hash-suffix for a name when syncing policies
+* remove `mesh` from Universal primary key
+
+## Decision Outcome
+
+Chosen option: use hash-suffix for a name when syncing policies
+
+### Implementation
+
+KDS has client and server, server owns the resource, client receive update when resource is changing. 
+Depending on the resource type Global and Zone CP both act as a client and as a server:
+* Policies are created on Global CP (KDS Server), Zone CP receives copies of these policies (KDS Client). 
+* Data plane proxies are created on Zone CP (KDS Server), Global CP receives copies of DPPs (KDS Client).
+
+When this feature is implemented, KDS Server is going to name resources differently. Resource name is constructed like:
+
+```go
+name := fmt.Sprintf("%s-%s", resourceName, hash(resourceMesh, resourceZone, resourceNamespace))
+```
+
+Since renaming is happening on the KDS Server, KDS Client doesn't need to know anything about hashes. 
+It will automatically remove old policies and create new policies. 
+
+Information that we're using to build hash has to be set as `labels` on the resource. For example, 
```

Comment:
I'm not sure I understand why/for exactly which cases `labels` is necessary. Is it so we can reconstruct this information when syncing Zone -> Global -> Zone?

Reply:
If you sync DPP from Zone to Global today, you'll have something like this in the Global:
```
zone-1.my-dpp.my-origin-ns
```
But with hash you'll get only:
```
my-dpp-{{hash(zone-1, my-origin-ns, mesh-1)}}
```

Since we don't want to lose this information (users need it, GUI needs it), we must save it as labels.